src/test/unit/DelegationUnit.t.sol

```diff
@@ -76,6 +76,18 @@ contract DelegationUnitTests is EigenLayerTestHelper {
         delegationContract.undelegate(address(this));
     }
 
+    function testUndelegateWithFrozenOperator(address undelegator) public fuzzedAddress(undelegator) {
+        cheats.startPrank(undelegator);
+        delegationContract.registerAsOperator(IDelegationTerms(address(this)));
+        cheats.stopPrank();
+
+        slasherMock.setOperatorFrozenStatus(address(this), true);
+        cheats.expectRevert(bytes("StrategyManager.onlyNotFrozen: staker has been frozen and may be subject to slashing"));
+        cheats.startPrank(address(strategyManagerMock));
+        delegationContract.undelegate(address(this));
+        cheats.stopPrank();
+    }
+
```

Comment:
move this test to the StrategyManager tests along with moving the modifier to the StrategyManager

Reply:
done